File: WebApplicationFSharp.Net/Areas/Customers/Controllers/CustomerController.fs
namespace WebApplicationFSharp.Net.Customers.Controllers

open System
open Microsoft.AspNetCore.Mvc
open Microsoft.FSharp.Linq.NullableOperators

open WebApplicationFSharp.Net.Customers.Models

[<Area("Customers")>]
type CustomerController () =
    inherit Controller()

    let _customers =
        [
            Customer (
                Id = Nullable 12,
                CustomerCode = "1001",
                CustomerName = "Asterix",
                NAS = "046454286",
                Amount = Nullable 90.34M,
                BirthDate = (Nullable <| DateTime(1008, 11, 01))
            )
            Customer (
                Id = Nullable 11,
                CustomerCode = "1002",
                CustomerName = "Obelix",
                NAS = "046454286",
                Amount = Nullable 101.00M,
                BirthDate = (Nullable <| DateTime(972, 10, 05))
            )
            Customer (
                Id = Nullable 0,
                CustomerCode = "0000",
                CustomerName = "",
                NAS = "000000000",
                Amount = Nullable 0.00M,
                BirthDate = (Nullable <| DateTime(0001, 01, 01))
            )
        ]

    // GET: /Customers/Customer/Index
    member this.Index () =
        this.View("CustomerForm") :> ActionResult

    // POST: /Customers/Customer/DisplayCustomer
    [<HttpPost>]
    member this.DisplayCustomer (obj: Customer) =
        match this.ModelState.IsValid with
        | true -> this.View("DisplayCustomer", obj) :> ActionResult
        | false -> this.ModelState.AddModelError("Customer Form", "There are errors in the form, please correct.") |> ignore
                   this.View("CustomerForm", obj) :> ActionResult

    // GET: /Customers/Customer/DisplayCustomer
    member this.DisplayCustomer (id) =
        let findDefault (customerList: Customer list) =
            customerList |> List.find (fun x -> x.Id ?= 0)
        let findById customerId (customerList: Customer list) =
            customerList |> List.tryFind (fun x -> x.Id ?= customerId)

        defaultArg (findById id _customers) (findDefault _customers) |> this.DisplayCustomer

    // GET: /Customers/Customer/IsAmountValide
    member this.IsAmountValide (nas: string, amount: Nullable<decimal>) =
        match nas with
        | null -> this.Json(true)
        | _ -> match amount with
               | x when x ?> 0.00M -> this.Json(true)
               | _ -> this.Json(false)

File: WebApplication.Net/WebApplication.Net/Areas/Customers/Controllers/CustomerController.cs
using System;
using System.Collections.Generic;
using Microsoft.AspNetCore.Mvc;

using WebApplication.Net.Customers.Models;

namespace WebApplication.Net.Customers.Controllers
{
    [Area("Customers")]
    public class CustomerController : Controller
    {
        private readonly IUrlHelper _urlHelper;
        private List<Customer> Customers = new List<Customer>();

        public CustomerController(IUrlHelper urlHelper)
        {
            _urlHelper = urlHelper;

            Customer obj1 = new Customer();
            obj1.Id = 12;
            obj1.CustomerCode = "1001";
            obj1.CustomerName = "Asterix";
            obj1.NAS = "046454286";
            obj1.Amount = 90.34M;
            obj1.BirthDate = new DateTime(1008, 11, 01);
            Customers.Add(obj1);

            obj1 = new Customer();
            obj1.Id = 11;
            obj1.CustomerCode = "1002";
            obj1.CustomerName = "Obelix";
            obj1.NAS = "046454286";
            obj1.Amount = 91.00M;
            obj1.BirthDate = new DateTime(972, 10, 05);
            Customers.Add(obj1);

            obj1 = new Customer();
            obj1.Id = 0;
            obj1.CustomerCode = "0000";
            obj1.NAS = "000000000";
            obj1.CustomerName = String.Empty;
            obj1.Amount = 0.00M;
            obj1.BirthDate = new DateTime(1, 1, 1);
            Customers.Add(obj1);
        }

        // GET: /Customers/Customer/Index
        public ActionResult Index()
        {
            return View("CustomerForm");
        }

        // POST: /Customers/Customer/DisplayCustomer
        [HttpPost]
        public ActionResult DisplayCustomer(Customer obj)
        {
            if (ModelState.IsValid)
            {
                obj.CustomerUrl = _urlHelper.Action("DisplayCustomer", "Customers", new { area = "Customer" });
                return View("DisplayCustomer", obj);
            }
            else
            {
                ModelState.AddModelError("Customer Form", "There are errors in the form, please correct.");
                return View("CustomerForm", obj);
            }
        }

        // GET: /Customers/Customer/DisplayCustomer
        public ActionResult DisplayCustomer(int id)
        {
            Customer objCustomer = Customers.Find(x => x.Id == id);
            if (objCustomer == null) objCustomer = Customers.Find(x => x.Id == 0);

            objCustomer.CustomerUrl = _urlHelper.Action("DisplayCustomer", "Customers", new { area = "Customer" });
            return DisplayCustomer(objCustomer);
        }

        // GET: /Customers/Customer/IsAmountValide
        public JsonResult IsAmountValide(Customer obj)
        {
            return (obj.NAS == null || obj.Amount > 0.0m)
                ? Json(true/*, JsonRequestBehavior.AllowGet*/)
                : Json(false/*, JsonRequestBehavior.AllowGet*/);
        }
    }
}
